.github/workflows/tests.yml

```diff
@@ -0,0 +1,91 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master, dev ]
+  pull_request:
+    branches: [ master, dev ]
+
+jobs:
+  tests:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]  #  TODO: Enable additional versions: 3.7
+        gcc-version: [8]  #  TODO: Enable additional versions: 6, 7
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Cache Conda
+      uses: actions/cache@v1
+      env:
+        # Increase this value to reset cache manually
+        CACHE_NUMBER: 0
+      with:
+        path: ~/conda_pkgs_dir
+        key:
+          ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('environment.yml') }}
+
+    - name: Cache pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/setup.py') }}
+
+    - name: Setup Conda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        environment-file: environment.yml
+        miniconda-version: "latest"
+        python-version: ${{ matrix.python-version }}
+        channel-priority: strict
+        use-only-tar-bz2: true  # IMPORTANT: This needs to be set for caching to work properly!
+
+    - name: Install OS dependencies
+      run: |
+        export GXX="g++-${{ matrix.gcc-version }}" GCC="gcc-${{ matrix.gcc-version }}"
+        sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
+        sudo apt-get -q update
+        sudo -E apt-get -yq --no-install-suggests --no-install-recommends install \
+            $GXX libxkbcommon-x11-0 x11-apps libgl1-mesa-glx qtbase5-dev
```

Comment:
I kind of remember there is a checklist for solving segfault problems with Qt.

Reply:
I am not sure whether `faulthandler` was enabled or not. Maybe you can try `pytest-faulthandler`.

Reply:
Btw, you are welcome to use https://github.com/zhujun98/foamlight for your test if `EXtra-foam` is too heavy.

Reply:
Awesome, thanks! Been pretty busy recently, I'll try to get back to this next week.